File: game-of-life-cpp/Fox.cpp
#include "include/Fox.h"

Fox::Fox(World* world, Point& position)
    : Animal(world, 3, 7, position, "🦊") {
}

Fox::Fox(World* world, Point& position, int age)
    : Animal(world, 3, 7, position, "🦊", age) {
}

Fox::~Fox() {
    
}

void Fox::action() {
    Point destination = world->findSafeSpaceAround(position);
    if(world->isWithinBounds(destination) && age != 0) {
        Organism* other = world->getAtCoordinates(destination);
    
        if(haveSavedAttack(*other)) {
            return;
        }
        other = world->getAtCoordinates(destination);
        int status = -1;
        if(other != nullptr) {
            status = collision(*other);
        }
        if (status != 1) move(destination);
    }

    age++;
}

File: game-of-life-java/src/Fox.java
import javax.swing.*;
import java.awt.*;

public class Fox extends Animal {
    private static final ImageIcon foxIcon = new ImageIcon("images/fox.png");
    public static final Image scaledFoxIcon = foxIcon.getImage().getScaledInstance(30, 30, Image.SCALE_SMOOTH);

    public Fox(Point position, World world, int age) {
        super(3, 7, position, world, age);
    }

    @Override
    public ImageIcon getIcon() {
        return foxIcon;
    }

    @Override
    public void move() {
        Point newPos = world.generateSafePosition(position);
        if (newPos != null) {
            Organism other = world.getOrganismAtPosition(newPos);
            if(haveSavedAttack(other)) {
                return;
            }
            other = world.getOrganismAtPosition(newPos);
            if (other != null) {
                collision(other);
            } else {
                position = newPos;
            }
        }
    }
}
